AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginFunctionalTests/GraphQLModelBasedTests+Concurrency.swift

```diff
@@ -11,75 +11,71 @@ import XCTest
 @testable import APIHostApp
 
 extension GraphQLModelBasedTests {
-    // TODO: Migrate this test to new async APIs
-    func testConcurrentSubscriptions() throws {
+    
+    func testConcurrentSubscriptions() async throws {
         let count = 50
-        let connectedInvoked = expectation(description: "Connection established")
-        connectedInvoked.expectedFulfillmentCount = count
-        let disconnectedInvoked = expectation(description: "Connection disconnected")
-        disconnectedInvoked.expectedFulfillmentCount = count
-        let completedInvoked = expectation(description: "Completed invoked")
-        completedInvoked.expectedFulfillmentCount = count
-        let progressInvoked = expectation(description: "progress invoked")
-        progressInvoked.expectedFulfillmentCount = count
+        let connectedInvoked = asyncExpectation(description: "Connection established", expectedFulfillmentCount: count)
+        let disconnectedInvoked = asyncExpectation(description: "Connection disconnected", expectedFulfillmentCount: count)
+        let completedInvoked = asyncExpectation(description: "Completed invoked", expectedFulfillmentCount: count)
+        let progressInvoked = asyncExpectation(description: "progress invoked", expectedFulfillmentCount: count)
+
         let uuid = UUID().uuidString
         let testMethodName = String("\(#function)".dropLast(2))
         let title = testMethodName + "Title"
 
-        let operations = AtomicValue<[GraphQLSubscriptionOperation<Post>]>(initialValue: [])
-        DispatchQueue.concurrentPerform(iterations: count) { _ in
-            let operation = Amplify.API.subscribe(
-                request: .subscription(of: Post.self, type: .onCreate),
-                valueListener: { event in
-                    switch event {
-                    case .connection(let state):
-                        switch state {
-                        case .connecting:
-                            break
-                        case .connected:
-                            connectedInvoked.fulfill()
-                        case .disconnected:
-                            disconnectedInvoked.fulfill()
-                        }
-                    case .data(let result):
-                        switch result {
-                        case .success(let post):
-                            if post.id == uuid {
-                                progressInvoked.fulfill()
+        let sequences = await withTaskGroup(of: AmplifyAsyncThrowingSequence<GraphQLSubscriptionEvent<Post>>.self) { group -> [AmplifyAsyncThrowingSequence<GraphQLSubscriptionEvent<Post>>] in
```

Comment:
This "concurrent subscription" test has been fixed after migrating the code to use TaskGroups, cc @brennanMKE

Reply:
👍